packages/react-core/src/components/ApplicationLauncher/examples/ApplicationLauncher.md

```diff
@@ -17,616 +17,50 @@ To add a tooltip, use the `tooltip` prop and optionally add more tooltip props b
-        favorites={favorites}
-        isGrouped
-      />
-    );
-  }
-}
+```ts file="./ApplicationLauncherFavoritesAndSearch.tsx"
 ```
 
 ### With custom icon
 
-```js
-import React from 'react';
-import { ApplicationLauncher, ApplicationLauncherItem } from '@patternfly/react-core';
-import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
-
-class ApplicationLauncheIcon extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isOpen: false
-    };
-    this.onToggle = isOpen => {
-      this.setState({
-        isOpen
-      });
-    };
-    this.onSelect = event => {
-      this.setState({
-        isOpen: !this.state.isOpen
-      });
-    };
-  }
-
-  render() {
-    const { isOpen } = this.state;
-    const appLauncherItems = [
-      <ApplicationLauncherItem key="application_1a" href="#">
-        Application 1 (anchor link)
-      </ApplicationLauncherItem>,
-      <ApplicationLauncherItem key="application_2a" component="button" onClick={() => alert('Clicked item 2')}>
-        Application 2 (button with onClick)
-      </ApplicationLauncherItem>,
-      <ApplicationLauncherItem key="disabled_application_4a" isDisabled>
-        Unavailable application
-      </ApplicationLauncherItem>
-    ];
-    return (
-      <ApplicationLauncher
-        onSelect={this.onSelect}
-        onToggle={this.onToggle}
-        isOpen={isOpen}
-        items={appLauncherItems}
-        toggleIcon={<HelpIcon />}
-      />
-    );
-  }
-}
+```ts file="./ApplicationLauncherCustomIcon.tsx"
 ```
 
 ### Basic with menu appended to document body
 
-```js
-import React from 'react';
-import { ApplicationLauncher, ApplicationLauncherItem } from '@patternfly/react-core';
-
-class ApplicationLauncherDocumentBody extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isOpen: false
-    };
-    this.onToggle = isOpen => {
-      this.setState({
-        isOpen
-      });
-    };
-    this.onSelect = event => {
-      this.setState({
-        isOpen: !this.state.isOpen
-      });
-    };
-  }
-
-  render() {
-    const { isOpen } = this.state;
-    const appLauncherItems = [
-      <ApplicationLauncherItem key="application_1a" href="#">
-        Application 1 (anchor link)
-      </ApplicationLauncherItem>,
-      <ApplicationLauncherItem key="application_2a" component="button" onClick={() => alert('Clicked item 2')}>
-        Application 2 (button with onClick)
-      </ApplicationLauncherItem>,
-      <ApplicationLauncherItem key="disabled_application_4a" isDisabled>
-        Unavailable application
-      </ApplicationLauncherItem>
-    ];
-    return (
-      <ApplicationLauncher
-        menuAppendTo={() => document.body}
-        onSelect={this.onSelect}
-        onToggle={this.onToggle}
-        isOpen={isOpen}
-        items={appLauncherItems}
-      />
-    );
-  }
-}
+```ts file="./ApplicationLauncherMenu.tsx"
```

Comment:
I see you've got some notable differences here between example names and component/file names. I can't remember if the team came to an alignment on this topic at the last meeting, unfortunately, but if not I feel like we should. I know that @thatblindgeye and I proposed a standard of `component name` + `example name` that matches 1:1, and I recall some debate on whether the name of the component was necessary but that's all I can recall at the moment.

I do understand (especially with the context of this example name) why you made the decisions you did though, and if we want to just standardize on something looser like `component name` + something that uniquely identifies the example I don't think I'd be opposed to that.

Reply:
Oh I see, I didn't realize we had said exactly the name of the example, versus something that resembles it (this leads to fewer characters in most cases), which is what I was aiming for. I'm fine with re-opening the discussion.

Reply:
I know that was our proposal, but I can't recall if there was a consensus reached on it and regretfully I didn't take notes. I do think some examples (like the one for the file I pointed out) have names that just don't work well for this purpose.

@thatblindgeye do you have any thoughts?

Reply:
I think componentName[exampleName] can work most of the time, but falling back to componentName[shortenedExampleName] would be good to have especially when it comes to something like "Basic with menu appended to document body". I think it's really just going to have to be a "use your best judgment" thing on which one to use and which best describes the example itself (in this instance, for example, I would say `ApplicationLauncherDocumentBody` or similar might be a more fitting name).

We could tweak the example name to something like "Menu appended to document body", but that would still result in a rather lengthy file/example name of `ApplicationLauncherMenuAppendedToDocumentBody` (give or take a few characters if this is shortened slightly further).

If we'd want the verbiage written down anywhere to actually reference I can write something a little more formal up as well, or help work on that.

Reply:
I agree that shortening the example part of the name would be good to keep the overall length of the filename down. What do you think of abbreviating the component name where applicable as well (AppLaunch or AppLauncher in this case), or dropping the component name section altogether because it isn't directly user displayed and can be inferred from the file structure?

Reply:
I had suggested the same as I thought the name of the component was redundant, but I don't recall there being much of a reception to that when discussed in standup a week ago.